File: src/main/java/com/dude/dms/ui/components/dialogs/TextBlockEditDialog.java
package com.dude.dms.ui.components.dialogs;

import com.dude.dms.backend.data.docs.TextBlock;
import com.dude.dms.backend.service.TextBlockService;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.radiobutton.RadioButtonGroup;
import com.vaadin.flow.component.radiobutton.RadioGroupVariant;
import com.vaadin.flow.component.textfield.TextField;

import java.util.ArrayList;
import java.util.List;

public class TextBlockEditDialog extends EventDialog {

    private final TextField text;

    private final RadioButtonGroup<String> group;

    private final TextBlock textBlock;

    private final String originalText;

    private final TextBlockService textBlockService;

    public TextBlockEditDialog(TextBlock textBlock, TextBlockService textBlockService) {
        this.textBlock = textBlock;
        this.textBlockService = textBlockService;

        originalText = textBlock.getText();

        setWidth("40vw");

        text = new TextField("Text");
        text.setWidthFull();
        text.setValue(originalText);

        group = new RadioButtonGroup<>();
        group.addThemeVariants(RadioGroupVariant.LUMO_VERTICAL);
        group.setItems(
                "change this",
                "change all in document (" + textBlockService.countByTextAndDoc(originalText, textBlock.getDoc()) + ')',
                "change all (" + textBlockService.countByext(originalText) + ')'
        );
        group.setValue("change this");

        Button createButton = new Button("Save", e -> save());
        createButton.setWidthFull();
        createButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);

        Button cancelButton = new Button("Close", e -> close());
        cancelButton.setWidthFull();
        cancelButton.addThemeVariants(ButtonVariant.LUMO_ERROR);

        HorizontalLayout buttonLayout = new HorizontalLayout(createButton, cancelButton);
        buttonLayout.setWidthFull();

        add(text, group, buttonLayout);
    }

    private void save() {
        if (text.isEmpty()) {
            text.setErrorMessage("Text can not be empty!");
            return;
        }
        String newText = text.getValue();
        textBlock.setText(newText);
        textBlockService.save(textBlock);
        List<TextBlock> textBlocks = new ArrayList<>();
        if (group.getValue().startsWith("change all in doc")) {
            textBlocks = textBlockService.findByTextAndDoc(originalText, textBlock.getDoc());
        } else if (group.getValue().startsWith("change all (")) {
            textBlocks = textBlockService.findByText(originalText);
        }
        textBlocks.forEach(block -> {
            block.setText(newText);
            textBlockService.save(block);
        });
        if (eventListener != null) {
            eventListener.onChange();
        }
        close();
    }
}

File: src/main/kotlin/com/dude/dms/ui/components/dialogs/TextBlockEditDialog.kt
package com.dude.dms.ui.components.dialogs

import com.dude.dms.backend.data.docs.TextBlock
import com.dude.dms.backend.service.TextBlockService
import com.vaadin.flow.component.button.Button
import com.vaadin.flow.component.button.ButtonVariant
import com.vaadin.flow.component.icon.VaadinIcon
import com.vaadin.flow.component.orderedlayout.HorizontalLayout
import com.vaadin.flow.component.radiobutton.RadioButtonGroup
import com.vaadin.flow.component.radiobutton.RadioGroupVariant
import com.vaadin.flow.component.textfield.TextField

class TextBlockEditDialog(private val textBlock: TextBlock, private val textBlockService: TextBlockService) : EventDialog<TextBlock>() {

    private val originalText = textBlock.text

    private val text = TextField("Text", originalText, "").apply { setWidthFull() }

    private val group = RadioButtonGroup<String>().apply {
        addThemeVariants(RadioGroupVariant.LUMO_VERTICAL)
        setItems(
                "change this",
                "change all in document (${textBlockService.countByTextAndDoc(originalText, textBlock.doc)})",
                "change all (${textBlockService.countByext(originalText)})"
        )
        value = "change this"
    }

    init {
        width = "40vw"

        val createButton = Button("Save", VaadinIcon.DISC.create()) { save() }.apply {
            setWidthFull()
            addThemeVariants(ButtonVariant.LUMO_PRIMARY)
        }
        val cancelButton = Button("Close", VaadinIcon.CLOSE.create()) { close() }.apply {
            setWidthFull()
            addThemeVariants(ButtonVariant.LUMO_ERROR)
        }
        add(text, group, HorizontalLayout(createButton, cancelButton).apply { setWidthFull() })
    }

    private fun save() {
        if (text.isEmpty) {
            text.errorMessage = "Text can not be empty!"
            return
        }
        val newText = text.value
        textBlock.text = newText
        textBlockService.save(textBlock)
        var textBlocks = emptyList<TextBlock>()
        if (group.value.startsWith("change all in doc")) {
            textBlocks = textBlockService.findByTextAndDoc(originalText, textBlock.doc)
        } else if (group.value.startsWith("change all (")) {
            textBlocks = textBlockService.findByText(originalText)
        }
        textBlocks.forEach {
            it.text = newText
            textBlockService.save(it)
        }
        triggerEditEvent(textBlock)
        close()
    }
}